specs/progress.md

```diff
@@ -0,0 +1,163 @@
+
+- Fin is scrolling a large list of continuously loading data as he scrolls he sees a indeterminate progress ring to let him know that data is loading.
+
+### Types
+*Progress* can be visually represented in 2 ways.
+- Linear - A straight horizontal line. `fast-progress`
+- Circular - A cicular ring shape. `fast-progress-ring`
+
+### Features
+
+A progress should allow the following attributes:
+- `value`, the value of the progress, if not passed the progress will be put into its "indeterminate" state.
+- `min`, the minimum value.
+- `max`, the maximum value.
+- `paused`, whether the progress is paused or not.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/progress)
+- [Material UI](https://material-ui.com/components/progress/)
+- [Lightning Design / Linear](https://www.lightningdesignsystem.com/components/progress-bar/)
+- [Lightning Design / Circular](https://www.lightningdesignsystem.com/components/progress-ring/)
+- [Ant Design](https://ant.design/components/progress/)
+- [Atlassian / Linear](https://atlaskit.atlassian.com/packages/server/progress-bar)
+- [Atlassian / Circular](https://atlaskit.atlassian.com/packages/core/spinner)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/progress-controls)
+
+---
+
+### API
+
+*Component name:*
+- `fast-progress`
+- `fast-progress-ring`
+
+*Attributes:*
+- `value`: number
+- `min`: number
+- `max`: number
+- `paused`: boolean
+
+*Slots:*
+- `indeterminate-linear`
+- `indeterminate-circular`
+
+### Anatomy and Appearance
+
+*Parts:*
+Linear parts:
+- determinate
+- indeterminate
+- linear-determinate
+- linear-indeterminate-indicator-1
+- linear-indeterminate-indicator-2
+
+Circular parts:
+- circular-container
+- circular-background
+- circular-determinate
+- circular-indeterminate
+
+*Circular Determinate Template:*
+```
+<div
+    role="progressbar"
+    aria-valuenow={50}
+    aria-valuemin={0}
+    aria-valuemax={100}
+>
+    <div part="determinate" slot={"determinate"}>
+        <svg part="circular-container" viewBox="0 0 16 16">
+            <circle part="circular-background" cx="8px" cy="8px" r="7px" />
+            <circle part="circular-determinate" style={{stroke-dasharray: "22px 44px"}} cx="8px" cy="8px" r="7px" />
+        </svg>
+    </div>
+</div>
+```
+
+*Circular Indeterminate Template:*
+```
+<div role="progressbar">
+    <div part="indeterminate" slot={"indeterminate"}>
+        <slot name="indeterminate-circular"></slot>
+            <svg part="circular-container" viewBox="0 0 16 16">
+                <circle part="circular-background" cx="8px" cy="8px" r="7px" />
+                <circle part="circular-indeterminate" cx="8px" cy="8px" r="7px" />
+            </svg>
+        </slot>
+    </div>
+</div>
+```
+
+*Linear Determinate Template:*
+```
+<div
+    role="progressbar"
+    aria-valuenow={50}
+    aria-valuemin={0}
+    aria-valuemax={100}
+>
+    <div part="determinate" slot={"determinate"}>
```

Comment:
minor nit, but you're mixing JSX into this - is that intentional?

Reply:
Yeah your right... Still getting used to this new "normal" word of css and straight html. ;)